src/server/game/AI/AIException.cpp

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2008-2018 TrinityCore <https://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AIException.h"
+
+InvalidAIException::InvalidAIException(char const* _message)
+{
+    message().assign(_message);
```

Comment:
do we really need a `.cpp` for this? `<string>` is visible in the header

Reply:
it always makes sense to declare in .h and then implement in .cpp

Reply:
it doesn't if you're adding a whole new compilation unit for no particular reason...